src/components/AttachmentCarousel.js

```diff
@@ -0,0 +1,241 @@
+            return;
+        }
+        this.setState(({attachments, page}) => {
+            const nextIndex = page + deltaSlide;
+            const {sourceURL, file} = this.getAttachment(attachments[nextIndex]);
+            this.props.onArrowPress({sourceURL, file});
+            return {
+                page: nextIndex,
+                sourceURL,
+                file,
+                isBackDisabled: nextIndex === 0,
+                isForwardDisabled: nextIndex === attachments.length - 1,
+            };
+        });
+    }
+
+    /**
+     * Listens for keyboard shortcuts and applies the action
+     *
+     * @param {Object} e
+     */
+    handleArrowPress(e) {
+        if (e.key === 'ArrowLeft') {
+            this.cycleThroughAttachments(-1);
+        }
+        if (e.key === 'ArrowRight') {
+            this.cycleThroughAttachments(1);
+        }
+    }
+
+    render() {
+        if (!this.state.sourceURL) {
+            return null;
+        }
+
+        return (
+            <View
+                style={[styles.attachmentModalArrowsContainer]}
+                onMouseEnter={() => this.onShowArrow(true)}
+                onMouseLeave={() => this.onShowArrow(false)}
+            >
+                {this.state.showArrows && (
+                    <>
+                        <Button
+                            medium
+                            style={[styles.leftAttachmentArrow]}
+                            icon={Expensicons.BackArrow}
+                            iconFill={themeColors.text}
+                            iconStyles={[styles.mr0, {PointerEvent: 'auto'}]}
+                            onPress={() => this.cycleThroughAttachments(-1)}
+                            isDisabled={this.state.isBackDisabled}
+                        />
+                        <Button
+                            medium
+                            style={[styles.rightAttachmentArrow]}
+                            icon={Expensicons.ArrowRight}
+                            iconFill={themeColors.text}
+                            iconStyles={[styles.mr0, {PointerEvent: 'auto'}]}
+                            onPress={() => this.cycleThroughAttachments(1)}
+                            isDisabled={this.state.isForwardDisabled}
+                        />
+                    </>
+                )}
+
+                <SwipeableView
+                    isAnimated
+                    styles={[styles.attachmentModalArrowsContainer]}
+                    canSwipeLeft={!this.state.isBackDisabled}
+                    canSwipeRight={!this.state.isForwardDisabled}
+                    onPress={() => this.canUseTouchScreen && this.onShowArrow(!this.state.showArrows)}
+                    onSwipeHorizontal={this.cycleThroughAttachments}
+                >
+                    <AttachmentView onPDFPress={() => this.onShowArrow(!this.state.showArrows)} sourceURL={this.state.sourceURL} file={this.state.file} />
```

Comment:
onShowArrow is called twice here

Reply:
It's actually only being used for the pdf because the native pdf component intercepts all gestures so I needed to drill down and handle the onpress there